src/display/editor/tools.js

```diff
@@ -817,6 +819,9 @@ class AnnotationEditorUIManager {
   }
 
   destroy() {
+    this.#updateModeCapability.resolve();
```

Comment:
Could this line perhaps explain the test failures? If so, the following should probably fix it.
```suggestion
    this.#updateModeCapability?.resolve();
```